lib/poller.js

```diff
@@ -146,19 +145,39 @@ function initializeEachSchema (schemas = ['public']) {
   }, schemas)
 }
 
+function recursiveTimeout (func, timeout, ...vars) {
```

Comment:
Honestly, I'm lost with both of these versions.

Reply:
Yeah, it used to be a simple setInterval, but that has the same pitfalls as the queue since interval doesn't wait for task completion. Ideally we want a real scheduler here, but for now we're just using recursive setTimeout, it changes the pattern a little bit, but it prevents multiple instances of the same job, and we don't really care about real time here.